nidl/callbacks/caching.py

```diff
@@ -0,0 +1,162 @@
+##########################################################################
+# NSAp - Copyright (C) CEA, 2025
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+from copy import deepcopy
+from typing import Any, Optional
+
+import pytorch_lightning as pl
+import torch
+import torch.nn as nn
+import torch.utils.data as data
+from pytorch_lightning import LightningModule, Trainer
+from torch.utils.data import DataLoader
+
+from nidl.utils import Bunch
+
+from ..estimators.base import BaseEstimator, TransformerMixin
+
+
+class CachingCallback(pl.Callback):
+    """ Cache embeddings.
+
+    This caching callback can be usefull in many situations such as model
```

Comment:
This caching callback should collect the outputs of all training_step/validation_step/predict_step/test_step using PL hooks (on_train_batch_end, on_val_batch_end, on_predict_batch_end, on_test_batch_end) as requested by the user and store them in a cache for quick computation of the metrics. I don't understand the current behavior of this callback.